apps/ui/src/suite/api/ServerGliaswapAPI.ts

```diff
@@ -385,31 +385,29 @@ export class ServerGliaswapAPI implements GliaswapAPI {
     };
   }
 
-  async searchSUDT(typeHash: string): Promise<CkbSudtAssetWithBalance | undefined> {
+  searchSUDT = async (typeHash: string): Promise<CkbAsset | undefined> => {
     const params = {
       typeHashOrAddress: typeHash,
     };
 
-    // @FIXME: replace to swap backend when it's ready
-    const res = await axios.get(`https://api-gliaswap.ckbapp.dev/tokens/search`, {
-      params,
-    });
+    const res = await this.axios.post(`/tokens/search`, params);
 
     if (isEmpty(res.data)) {
       return undefined;
     }
 
-    const [asset] = res.data;
+    const [{ info: asset }] = res.data;
 
     return {
       ...asset,
-      decimals: parseInt(asset.decimal, 10),
-      balance: asset.amount,
+      decimals: asset.decimal ? parseInt(asset.decimal, 10) : 8,
```

Comment:
It doesn't make sense that the decimal of all unknown tokens is 8

Reply:
Unknown token's decimal is unknown (null), so 8 is a random number.